core/src/main/java/org/apache/spark/memory/MemoryConsumer.java

```diff
@@ -130,4 +130,22 @@ protected void freePage(MemoryBlock page) {
     used -= page.size();
     taskMemoryManager.freePage(page, this);
   }
+
+  /**
+   * Allocates a heap memory of `size`.
+   */
+  public long allocateHeapExecutionMemory(long size) {
```

Comment:
This function does not actually create any object, I'd like to call it `acquireOnHeapMemory`